xarray/tests/test_conventions.py

```diff
@@ -333,6 +333,16 @@ def test_invalid_time_units_raises_eagerly(self) -> None:
         with pytest.raises(ValueError, match=r"unable to decode time"):
             decode_cf(ds)
 
+    def test_invalid_timedelta_units_do_not_decode(self) -> None:
+        # regression test for #8269
+        ds = Dataset(
+            {"time": ("time", [0, 1, 20], {"units": "days invalid", "_FillValue": 20})}
+        )
+        expected = Dataset(
+            {"time": ("time", [0.0, 1.0, np.nan], {"units": "days invalid"})}
+        )
+        assert_identical(expected, decode_cf(ds))
```

Comment:
Could you parametrize this test to make sure this works for both the `decode_times=True` and `decode_times=False` cases?